### DISCUSSION TITLE
Ubuntu 22.04 LTS

### DISCUSSION THREAD

**Discussion by Commaster** [accetto/ubuntu-vnc-xfce-g3#30] [General]
Any plans to update the base image to the next LTS?

**Comment by Commaster** [comment]
I'd also appreciate if it had the _sane_ version of Firefox (https://www.omgubuntu.co.uk/2022/04/how-to-install-firefox-deb-apt-ubuntu-22-04):
```
$ sudo snap remove firefox
$ sudo add-apt-repository ppa:mozillateam/ppa
$ echo '
Package: *
Pin: release o=LP-PPA-mozillateam
Pin-Priority: 1001
' | sudo tee /etc/apt/preferences.d/mozilla-firefox
$ sudo apt install firefox
```

**Comment by accetto** [comment]
Hello @Commaster , thanks a lot for your feedback.

Actually I wanted to port the images to `22.04 LTS` right in April, but I've been stopped by the badly suprising Firefox's decision to deliver only the snappy version. I've never got `snap` working in a Docker container, because of its dependency on `systemd`. I've hoped, that [sysbox](https://github.com/nestybox/sysbox) would help, but not luck yet.

Personally I don't like the Firefox's decision so much, that I've switched from `Ubuntu` to `Debian` and I plan to relase a `Debian` edition of the images. :)

I'll test your snippet and if it'll work, I'll test porting to `Ubuntu 22.04 LTS`.

However, do you know how much stuff in `Ubuntu 22.04 LTS` actually depends on `snap`? Is it worth porting? Isn't switching to `Debian` a better idea? I would appreciate your opinion very much.

Regards,
accetto

**Reply by Commaster** [reply]
I've also been frustrated by the snap migration of Firefox, mainly because it breaks geckodriver that I use every day. Luckily I was able to google that guide and their "official" ppa right away and upgrade without issues.

Currently my desktop PC has nothing installed in `snap list` (not even snapd itself) and I could uninstall the snapd package to free up 100 MB disk space.

**Reply by Commaster** [reply]
P.S. You could also (possibly) simplify the containers a bit, since docker has built-in `tini` now: https://docs.docker.com/engine/reference/run/#specify-an-init-process

(I previously used https://github.com/fcwu/docker-ubuntu-vnc-desktop but that doesn't seem to be active and no hope to get a 22.04 update there)

**Reply by accetto** [reply]
You're right about `tini`, it has been copied from the older generation. Thanks for the hint, I'll remove it.

Before I'll begin with porting to `Ubuntu 22.04 LTS`, would you have, maybe, a similar tip also for `Chromium Browser`?

As a quick solution I've originally used the `18.04` edition and since then I never had time to test a non-snappy installation. A proved snippet would really help.

Update on 2023-01-18:

I have to correct myself about `tiny`. I've tested it and I've decided not to remove it. It's true, that it's possible to use `docker run --init`, but if the user forgets it, then the `startup.sh` becomes `pid 1`. However, the startup script is not designed to fullfill the usual responsibilities of an init system.

**Reply by Commaster** [reply]
According to quick googling, there is a way (https://ubuntuhandbook.org/index.php/2022/05/install-chromium-deb-ubuntu-2204/), though not tested by me personally:

```
$ sudo add-apt-repository ppa:savoury1/chromium
$ echo '
Package: *
Pin: release o=LP-PPA-savoury1
Pin-Priority: 1001
' | sudo tee /etc/apt/preferences.d/savoury1-chromium
$ sudo apt install chromium-browser
```

Please note that both the previous snippet and this one are from the mentioned links, not invented by me.

**Reply by accetto** [reply]
Thanks for the hint. However, I would prefer an official PPA, so I have to check it first.

**Reply by Commaster** [reply]
Alternatively you could use Chrome, which has an "official" repository, as explained here: https://www.ubuntuupdates.org/ppa/google_chrome

**Reply by accetto** [reply]
Thanks, could be also an option. However, I have a `Chrome` image in the previous generagion set (G2) and there never been many pulls. Seems like people prefer `Chromium` instead.

**Comment by accetto** [comment]
Images based on `Ubuntu 22.04 LTS` have been just released (see `Release 23.01`).

**Comment by accetto** [comment]
Images based on `Debian 11 (bullseye)` have been just released (see [accetto/debian-vnc-xfce-g3](https://github.com/accetto/debian-vnc-xfce-g3)).

**Comment by sacp175** [comment]
Thanks lot for this image, was facing of issues for browser installation on base [Ubuntu 22.04 LTS] image.

Want to check novnc client connection  to VNCServer of remote device on internet.

I guess default novnc client connect to localhost VNCServer. 
-------------------------------------------------------------------------------------------------
python3 -m websockify --web /usr/libexec/noVNCdim/utils/../ 6901 localhost:5901
-------------------------------------------------------------------------------------------------
How to change this localhost to other ip address of remote device in image.

Thanks in advance.

**Reply by accetto** [reply]
I'm not sure if  I understand your question correctly. What are you trying to achieve?

If you want a container with Chromium or Firefox browsers, I already provide several of them. You can access them via VNC or noVNC on any IP address, that your host computer exposes.

For example, if you create the container like this:

```shell
docker run -p "36901:6901" ...
```

then you can access the container on any host interface, because the port will be exposed like `0.0.0.0:36901`.

You can connect to it like `http://127.0.0.1:36901`, but also `http://A.B.C.D:36901`, where `A.B.C.D` is any of the IP addresses on your host.

Is that what you wanted?

**Reply by sacp175** [reply]
@accetto 
Thanks for reply. Sorry I could not mention properly. Added below image and description.

![image](https://user-images.githubusercontent.com/72434602/234147180-389e3b7e-603f-4d63-8d52-c63766042074.png)

1. PC1 take remote desktop of PC2 container --> Okay
2. PC2 container[novnc] take remote desktop of PC3[vncserve]--> Here want to configure PC3 ip address in novnc of PC container

Please confirm.

**Reply by accetto** [reply]
Oh, that's interesting. However, what's the real goal? Is it required, that the PC1 user can control the PC3? Why do you want to have a container in between?

Maybe you can also ask the authors of the `TigerVNC` and `noVNC` software? They will be able to answer it better, I think. 
I would need to do some experimenting first.

**Reply by sacp175** [reply]
It worked after updating VNC_DEST to PC3 IP in /usr/libexec/noVNCdim/utils/novnc_proxy.
Thanks for your support and image.
